edb/lang/edgeql/compiler/clauses.py

```diff
@@ -71,6 +75,33 @@ def compile_orderby_clause(
                 ir_sortexpr.context = sortexpr.context
                 stmtctx.enforce_singleton(ir_sortexpr, ctx=exprctx)
 
+                # check that the sortexpr type is actually orderable
+                sort_type = inference.infer_type(ir_sortexpr, exprctx)
+                # Verify that a comparison operator is defined for 2
+                # sort_type expressions.
+                env = exprctx.env
```

Comment:
Comments don't really correspond to code here.  "Check that" and you only infer type; "verify that" and you alias an attribute, etc.  Please review.